### PR TITLE
Upgrade to webpack 5

### PR DESCRIPTION
**Pull request by rchiodo**
Fixes https://github.com/microsoft/vscode-jupyter/issues/7827

### CODE REVIEW

File: build/webpack/webpack.datascience-ui.config.builder.js [hunk 1]
```diff
@@ -235,17 +234,16 @@ function buildConfiguration(bundle) {
...
                 {
                     test: /\.css$/,
-                    use: ['cache-loader', 'thread-loader', 'style-loader', 'css-loader']
+                    use: ['thread-loader', 'style-loader', 'css-loader']
                 },
                 {
                     test: /\.js$/,
                     include: /node_modules.*remark.*default.*js/,
                     use: [
-                        'cache-loader',
```
Comment (rchiodo) [inline build/webpack/webpack.datascience-ui.config.builder.js:248]: Cache loader not supported in Webpack 5

File: src/client/datascience/constants.ts [hunk 2]
```diff
@@ -2,557 +2,10 @@
 // Licensed under the MIT License.
 'use strict';
 
-import * as path from 'path';
 import { EXTENSION_ROOT_DIR, JVSC_EXTENSION_ID } from '../common/constants';
+import * as path from 'path';
 
-export const DefaultTheme = 'Default Light+';
```
Comment (rchiodo) [inline src/client/datascience/constants.ts:8]: Moved all of these to datascience-ui/constants so they can be referenced in the UI parts without referencing node constructs (webpack no longer auto fills all node methods)

File: src/datascience-ui/interactive-common/redux/store.ts [hunk 3]
```diff
@@ -159,6 +156,7 @@ function createMiddleWare(
     const testMiddleware =
         forceOnTestMiddleware || testMode || isUITest ? createTestMiddleware(transformLoad) : undefined;
 
+    /* Fixup this code if you need to debug redux
```
Comment (IanMatthewHuff) [inline src/datascience-ui/interactive-common/redux/store.ts:159]: I think some of our tests rely on this middleware logging still.

Comment (IanMatthewHuff) [reply src/datascience-ui/interactive-common/redux/store.ts:159]: I guess we'll see when the test runs come in, but I think some of the variable tests wait for variable complete logging from the middleware.

Comment (rchiodo) [reply src/datascience-ui/interactive-common/redux/store.ts:159]: From what I know, the process environment variables were never set to anything, so unless I misread the logic here, the testMiddleware isn't even possible to enable.

Comment (IanMatthewHuff) [reply src/datascience-ui/interactive-common/redux/store.ts:159]: Hmm, I think I was thinking of the forceTestMiddleware part of webview.ts. Does that still work? Pretty sure that was checking ENV.

Comment (IanMatthewHuff) [reply src/datascience-ui/interactive-common/redux/store.ts:159]: Ahh, that's still on the main node side. Nevermind.

File: package.json [hunk 4]
```diff
@@ -2307,11 +2306,13 @@
         "node-gyp-build": "^4.2.3",
         "node-has-native-dependencies": "^1.0.2",
         "node-html-parser": "^1.1.13",
+        "node-loader": "^2.0.0",
```
Comment (IanMatthewHuff) [inline package.json:2309]: Is this used? I don't see it added in the webpack files.

**Review by IanMatthewHuff — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (rchiodo) [reply package.json:2309]: Oh I don't need it. Let me remove.

**Pull request merged**